apis/druid/v1alpha1/druidingestion_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+DataInfra 2023 Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DruidIngestionMethod string
+
+const (
+	Kafka                    DruidIngestionMethod = "kafka"
+	Kinesis                  DruidIngestionMethod = "kinesis"
+	NativeBatchIndexParallel DruidIngestionMethod = "native-batch"
+	QueryControllerSQL       DruidIngestionMethod = "sql"
+	HadoopIndexHadoop        DruidIngestionMethod = "index-hadoop"
+)
+
+type DruidIngestionSpec struct {
+	// +optional
+	Suspend bool `json:"suspend"`
+	// +required
+	DruidClusterName string `json:"druidCluster"`
```

Comment:
Is cluster name enough? Why not `v1.ObjectReference` to also support different namespaces?

Reply:
CR is scoped to namespace. For consistency want to force user to create ingestion spec in same namespace as CR.